battle.cpp

```diff
@@ -377,7 +374,7 @@ void AddBattleFact(
 ) {
     if (!Globals->WORLD_EVENTS) return;
 
-    auto fact = new BattleFact;
+    BattleFact * fact = new BattleFact();
```

Comment:
this is redundant with my currently up PR.. I don't care about BattleFact * vs auto here, and the explicit is better in this case, so when resolving the inevitable merge conflict, will go with that.